api/src/controllers/validation.ts

```diff
@@ -0,0 +1,53 @@
+import { Validator as FluentValidator } from 'fluentvalidation-ts';
+import { validate as uuidValidate } from 'uuid';
+
+import ValidationException from '../exceptions/ValidationException';
+
+/**
+ * Validator class that adds an extra method to the fluentvalidation-ts validator that
+ * throws a ValidationException.
+ */
+class Validator<TModel> extends FluentValidator<TModel> {
```

Comment:
What's `TModel`?

Reply:
`TModel` is the generic type that a `Validator` class will validating. For example it might be `ReqBody` or `ReqQuery` or `ReqParam` etc.

This section of the docs would be helpful for you to read I think https://fluentvalidation-ts.alexpotter.dev/docs/overview.html#the-gist